utility/importer/brave_importer.cc

```diff
@@ -0,0 +1,440 @@
+    return;
+  }
+
+  std::unique_ptr<base::Value> session_store_json =
+    base::JSONReader::Read(session_store_content);
+  if (!session_store_json) {
+    LOG(ERROR) << "Parsing Brave session store JSON failed";
+    return;
+  }
+
+  base::Value* bookmark_folders_dict =
+    session_store_json->FindKeyOfType("bookmarkFolders",
+                                      base::Value::Type::DICTIONARY);
+  base::Value* bookmarks_dict =
+    session_store_json->FindKeyOfType("bookmarks",
+                                      base::Value::Type::DICTIONARY);
+  base::Value* bookmark_order_dict =
+    session_store_json->FindPathOfType({"cache", "bookmarkOrder"},
+				       base::Value::Type::DICTIONARY);
+  if (!(bookmark_folders_dict && bookmarks_dict && bookmark_order_dict))
+    return;
+
+  // Recursively load bookmarks from each of the top-level bookmarks
+  // folders: "Bookmarks Toolbar" and "Other Bookmarks"
+  std::vector<base::string16> path;
+  RecursiveReadBookmarksFolder(base::UTF8ToUTF16("Bookmarks Toolbar"),
+                               "0",
+                               path,
+                               true,
+                               bookmark_folders_dict,
+                               bookmarks_dict,
+                               bookmark_order_dict,
+                               bookmarks);
+
+  RecursiveReadBookmarksFolder(base::UTF8ToUTF16("Other Bookmarks"),
+                               "-1",
+                               path,
+                               false,
+                               bookmark_folders_dict,
+                               bookmarks_dict,
+                               bookmark_order_dict,
+                               bookmarks);
+}
+
+void BraveImporter::RecursiveReadBookmarksFolder(
+  const base::string16 name,
+  const std::string key,
+  std::vector<base::string16> path,
+  const bool in_toolbar,
+  base::Value* bookmark_folders_dict,
+  base::Value* bookmarks_dict,
+  base::Value* bookmark_order_dict,
+  std::vector<ImportedBookmarkEntry>* bookmarks) {
+  // Add the name of the current folder to the path
+  path.push_back(name);
+
+  base::Value* bookmark_order =
+    bookmark_order_dict->FindKeyOfType(key, base::Value::Type::LIST);
+  if (!bookmark_order)
+    return;
+
+  for (const auto& entry : bookmark_order->GetList()) {
+    auto& type = entry.FindKeyOfType("type", base::Value::Type::STRING)->GetString();
+    auto& key = entry.FindKeyOfType("key", base::Value::Type::STRING)->GetString();
+
+     if (type == "bookmark-folder") {
+      base::Value* bookmark_folder =
+        bookmark_folders_dict->FindKeyOfType(key, base::Value::Type::DICTIONARY);
+      auto& title =
+        bookmark_folder->FindKeyOfType("title", base::Value::Type::STRING)->GetString();
+
+      // Empty folders don't have a corresponding entry in bookmark_order_dict,
+      // which provides an easy way to test whether a folder is empty.
+      base::Value* bookmark_order_entry =
```

Comment:
this seems weird to me because empty bookmark folder still requires order. (ex. you can rearrange their position)
Also, I verified it does exist in `bookmarkOrder` with key equals to folderId